src/main/java/tutorspet/logic/util/AttendanceRecordListUtil.java

```diff
@@ -23,6 +25,55 @@
  */
 public class AttendanceRecordListUtil {
 
+    /**
+     * Returns an {@code AttendanceRecordList} where the {@code studentToRemove} has been removed.
+     */
+    public static AttendanceRecordList removeStudentFromAttendanceRecordList(
+            AttendanceRecordList targetAttendanceRecordList, Student studentToRemove) {
+        requireAllNonNull(targetAttendanceRecordList, studentToRemove);
+
+        List<AttendanceRecord> attendanceRecords = targetAttendanceRecordList.getAttendanceRecordList();
+
+        List<AttendanceRecord> updatedAttendanceRecords =
+                attendanceRecords.stream().map(attendanceRecord ->
+                        removeStudentFromAttendanceRecord(attendanceRecord, studentToRemove))
+                        .collect(Collectors.toUnmodifiableList());
+
+        return new AttendanceRecordList(updatedAttendanceRecords);
+    }
+
+    /**
+     * Returns an {@code AttendanceRecordList} where all {@code Student}s has been removed.
+     */
+    public static AttendanceRecordList removeAllStudentsFromAttendanceRecordList(
+            AttendanceRecordList targetAttendanceRecordList) {
+        requireAllNonNull(targetAttendanceRecordList);
+
+        List<AttendanceRecord> attendanceRecords = targetAttendanceRecordList.getAttendanceRecordList();
+
+        List<AttendanceRecord> updatedAttendanceRecords =
+                attendanceRecords.stream().map(attendanceRecord -> new AttendanceRecord())
+                        .collect(Collectors.toUnmodifiableList());
+
+        return new AttendanceRecordList(updatedAttendanceRecords);
+    }
+
+    /**
+     * Returns an {@code AttendanceRecord} where the {@code studentToRemove} has been removed.
+     */
+    private static AttendanceRecord removeStudentFromAttendanceRecord(
+            AttendanceRecord attendanceRecord, Student studentToRemove) {
+        if (!attendanceRecord.hasAttendance(studentToRemove.getUuid())) {
+            return attendanceRecord;
+        }
+
+        try {
```

Comment:
I'd like to check with you guys on how we should go about handling this block.

Problem : ```removeAttendance(...)``` throws ```CommandException``` when it is unable to find an attendance to delete. This works for individual attendances, but if we doing a cascading delete (ie removing all occurrences of the student), then we would want to return the ```attendanceRecord``` even if no student can be deleted.

Solution 1 (Currently implemented) : Do a try and catch to overwrite the ```CommandException```

Solution 2 (Can consider) : Introduce a boolean called ```isCascadingRemoval``` into ```removeAttendance(...)```. Then don't throw any exceptions if its a cascading removal, and instead return the attendanceRecord. Downside is that will need to put in extra manhours to refactor the test cases for ```removeAttendance(...)``

Solution 3 : Remove the ```removeAttendance(...)``` abstraction and use the underlying code in ```removeAttendance(...)``` in this method. But this will violate SRP.

Solution 4 : Propagate the ```CommandException``` upwards to ```LessonUtil``` and ```ModuleClassUtil``` and eventually ```UniqueModuleClassList```. But downside is that you eventually have to handle it with try catch

Reply:
To go with solution 1